File: isma-intg-lib/isma-intg-lib-rkfehlberg/src/main/java/ru/nstu/isma/intg/lib/rungeKutta/rungeKuttaFehlberg/internal/RkFehlbergStage2Calculator.java
package ru.nstu.isma.intg.lib.rungeKutta.rungeKuttaFehlberg.internal;

import ru.nstu.isma.intg.api.methods.*;

public class RkFehlbergStage2Calculator extends StageCalculator {
	@Override
	public double yk(final double step, final double y, final double f, final double[] stages) {
		return y + 0.25 * stages[0];
	}

	@Override
	public double k(final double step, final double y, final double f, final double[] stages, final double stagesY,
			final double stagesF) {
		return step * stagesF;
	}
}

File: isma-intg-lib/isma-intg-lib-rkfehlberg/src/main/kotlin/ru/nstu/isma/intg/lib/rungeKutta/rungeKuttaFehlberg/internal/RkFehlbergStage2Calculator.kt
package ru.nstu.isma.intg.lib.rungeKutta.rungeKuttaFehlberg.internal

import ru.nstu.isma.intg.api.methods.StageCalculator

class RkFehlbergStage2Calculator : StageCalculator() {
    override fun yk(step: Double, y: Double, f: Double, stages: DoubleArray): Double {
        return y + 0.25 * stages[0]
    }

    override fun k(
        step: Double, y: Double, f: Double, stages: DoubleArray, stagesY: Double,
        stagesF: Double
    ): Double {
        return step * stagesF
    }
}
